### ISSUE TITLE
pbi-tools 1.0 rc8 creating invalid PBIT files

### ISSUE THREAD

**Issue by magnusneck** [pbi-tools/pbi-tools#325]
The compile and deploy actions in rc8 creates invalid PBIT/PBIX files for import model reports when the source files are in TMDL format. 

When opening the compiled PBIT file I get the following error in Power BI desktop:

Cannot convert value '' to required type 'String'. Check path 'name', line 1, position 12.

Using pbi-tools.1.0.0-rc.5+preview.1 I get a valid PBIT file. Comparing the compiled PBIT files I notice that DataModelSchema file contains a difference between rc.5 and rc.8. The initial attribute "name" is null in rc.8 but "Model" in rc.5+preview.1.

When using the deploy action I get the similar error in a pipeline:

[14:03:48.762 DBG] [PbiTools.Deployments.DeploymentManager] Deserializing tabular model from sources...
[14:03:48.969 FTL] [] An unhandled exception occurred.
System.AggregateException: One or more errors occurred. (Cannot convert value '' to required type 'String'. Check path 'name', line 2, position 14.)
 ---> Microsoft.AnalysisServices.JsonSerializationException: Cannot convert value '' to required type 'String'. Check path 'name', line 2, position 14.
   at Microsoft.AnalysisServices.Tabular.Json.JsonPropertyHelper.ConvertJsonValueToString(JToken rawJsonValue)
   at Microsoft.AnalysisServices.Tabular.Database.ReadPropertyFromJson(JProperty jsonProp, DeserializeOptions options, CompatibilityMode mode)
   at Microsoft.AnalysisServices.Tabular.Database.DeserializeFromJsonObject(JObject jsonObj, DeserializeOptions options, CompatibilityMode mode)
   at Microsoft.AnalysisServices.Tabular.JsonSerializer.DeserializeDatabase(String json, DeserializeOptions options, CompatibilityMode mode)
   at PbiTools.Deployments.DeploymentManager.DeployDatasetAsync(PbiDeploymentManifest manifest, String label, String environment) in X:\pbi-tools\BRANCHES\rc.8\src\PBI-Tools\Deployments\DeploymentManager.Dataset.cs:line 128
   at PbiTools.Deployments.DeploymentManager.DeployAsync(String profileName, String environment) in X:\pbi-tools\BRANCHES\rc.8\src\PBI-Tools\Deployments\DeploymentManager.cs:line 91
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Task.Wait()
   at PbiTools.Cli.CmdLineActions.Deploy(String folder, String label, String environment, String basePath, Boolean whatIf) in X:\pbi-tools\BRANCHES\rc.8\src\PBI-Tools\Cli\Deploy.cs:line 41

Switching to legacy serialization model solves the problem.

**Comment by mthierba** [comment]
Thanks for the detailed error report. I'll get that fixed for the upcoming release.

**Comment by mthierba** [comment]
I'm assuming you have a TMDL model? Give your database in `database.tmdl` a (any) name, similar to here: https://github.com/pbi-tools/sales-sample/commit/44e9c03f86efd0ef0f61dd909226be2c60f38543#diff-aa27fd66a7234bfb81821df9cdd72c68ba12d2a8b8fa39716d0345a6f041cbec to resolve the issue.

**Comment by toseekandfind** [comment]
None of my extracted PBI files have a database.tmdl file.  I'd be happy to hop on a call if you'd like or continue to discuss here. 

When using a previous version (rc.5+preview.1), the files are valid, suggesting a regression in rc8 and the releases between rc.5 to rc8. 

We're upgrading from rc.4 and rc.5 is the latest stable release that doesn't produce this error. I've tried each version in between to see if this error is produced and rc.5 is the latest stable version that doesn't produce it.

**Comment by toseekandfind** [comment]
I believe you are moving development in collaboration with Microsoft to the .tmdl file types in the serialization but coming from an earlier version of pbi-tools and finally having the time to tackle some tech debt I haven't been following this very closely. Is there something we need to do like forcefully do a conversion?

**Comment by mthierba** [comment]
Could you send a screenshot of the first few lines of your `Model/database.json` file? And please confirm which PBI Desktop version you use.

**Comment by toseekandfind** [comment]
Seems like the issue might be the .tmd file type was used for compiles/extracts in prior versions(rc.4) and now is .tmdl. When directly upgrading and trying to compile files from our repo this happens likely because of the format. pbi-tools doesn't pick up on this in a backward compatible way.

As next steps i'm thinking we will need to do manual extracts of all our .pbix files using the newer version of pbi-tools then compiling them once they've been added to the repo in the .tmdl file type might not be an issue when we open the reports.

Might be able to do this with a convert but haven't used that functionality yet with pbi-tools.

**Comment by mthierba** [comment]
Whilst TMDL is in public preview (by Microsoft, not pbi-tools), it's always a good idea to keep a backup of the BIM file for the model, or temporarily convert to BIM when switching between pbi-tools versions:

https://pbi.tools/tmdl/#convert-a-model-folder-tmdl-pbixproj-to-bim-file

________________________________

**Comment by toseekandfind** [comment]
While implementing I noticed an issue related to this. 

It looks like rc8 doesn't create linage tags. This then causes issues in compile/extracts. I'll post a screenshot below. On the left is rc8 on the right is rc7.:

![Version 8 vs Version 7 side by side comparison](https://github.com/pbi-tools/pbi-tools/assets/45746031/035db7f2-8e19-4de0-af24-4d6e20d57543)

**Comment by mthierba** [comment]
@toseekandfind `pbi-tools` is merely using Microsoft's APIs to read & write TMDL - it doesn't interfere with the contents of those files.
Specifically, rc7 is using the TOM libraries 19.69, whereas rc8 uses 19.74 - there may well be a breaking change in Microsoft's client libraries. Whether or not a model contains lineage tags also depends on where and how to model was authored. Power BI Desktop, for instance, always emits those on any models created by Desktop.
Long story short - this is out of scope for `pbi-tools`.
